numpy/linalg/_linalg.py

```diff
@@ -3396,7 +3396,12 @@ def matrix_transpose(x, /):
     return _core_matrix_transpose(x)
 
 
-matrix_transpose.__doc__ = _core_matrix_transpose.__doc__
+matrix_transpose.__doc__ = _core_matrix_transpose.__doc__ + ("""
+
+    Notes
+    -----
+    This function is an alias of `numpy.matrix_transpose`.
+""")
```

Comment:
`TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'` 🫠 what is the appropriate fix here?
```suggestion
matrix_transpose.__doc__ = (_core_matrix_transpose.__doc__ + ("""

    Notes
    -----
    This function is an alias of `numpy.matrix_transpose`.
""")) if _core_matrix_transpose.__doc__ else None
```

Reply:
This is surprising that for one job `__doc__` is `None`. No idea why it happens, and it shouldn't. One way to circumvent it you can make it:
```python
matrix_transpose.__doc__ = f"""{_core_matrix_transpose.__doc__}
    Notes
    -----
    This function is an alias of `numpy.matrix_transpose`.
"""
```
It works for `None` and strings.

But to be honest I incline towards writing `np.linalg.matrix_transpose` and actual docstring, where the Notes is in the right place (or rather where usual, before Examples), and Examples use the correct function.

This is a bit of docstring repetition, but I think we already do it in multiple places for different functions:
```python
@array_function_dispatch(_matrix_transpose_dispatcher)
def matrix_transpose(x, /):
    """
    Transposes a matrix (or a stack of matrices) ``x``.

    This function is Array API compatible.

    Parameters
    ----------
    x : array_like
        Input array having shape (..., M, N) and whose two innermost
        dimensions form ``MxN`` matrices.

    Returns
    -------
    out : ndarray
        An array containing the transpose for each matrix and having shape
        (..., N, M).

    See Also
    --------
    transpose : Generic transpose method.

    Notes
    -----
    This function is an alias of :func:`numpy.matrix_transpose`.

    Examples
    --------
    >>> import numpy as np
    >>> np.linalg.matrix_transpose([[1, 2], [3, 4]])
    array([[1, 3],
           [2, 4]])
           
    """
    return _core_matrix_transpose(x)
```

Reply:
> where the Notes is in the right place (or rather where usual, before Examples),

I don't think this is too big a deal, as numpydoc automatically reorders things anyway (see the screenshot in PR description)

> Examples use the correct function

That would be nice, although I don't think it matters much with the alias note right above it

Reply:
>I don't think this is too big a deal, as numpydoc automatically reorders things anyway

Thanks! Then I guess you can use string interpolation, so this `None` in one test environment doesn't cause trouble (or provide full docstring, either is fine IMO)

Reply:
```suggestion
matrix_transpose.__doc__ = f"""{_core_matrix_transpose.__doc__}

    Notes
    -----
    This function is an alias of `numpy.matrix_transpose`.
"""
```